### ISSUE TITLE
Migrate development & staging websites to gov.uk domains

### ISSUE THREAD

**Issue by Ben-Hodgkiss** [digital-land/digital-land#352]
**Overview**
We have been notified that the deevelopment and staging environments for the platform have been flagged by the [NCSC Takedown Service](https://www.ncsc.gov.uk/information/takedown-service) as they use the Government branding on a non-gov.uk domain. This has also restricted users from being able to access these domains from some machines and has warnings for other users when accessing.

In line with the [Service Manual](https://www.gov.uk/service-manual/technology/managing-domain-names), we should migrate these domains from digital-land.info to subdomains of planning.data.gov.uk.

**Technical Approach**

- [x] Gain understanding of existing DNS setup for planning.data.gov.uk domain and document in technical-documentation Wiki
- [x] Determine whether there are any manual references to digital-land.info domains in Terraform code and refactor where possible
- [x] Terraform code for development Route 53 zone
- [x] Terraform code for staging Route 53 zone
- [x] Make any manual configuration changes as necessary and document in technical-documentation Wiki

**Comment by cpcundill** [comment]
Hey @eveleighoj  - as expected, this is quite a tricky change.  Here's a PR with the changes so far:

https://github.com/digital-land/digital-land-infrastructure/pull/191

Here's my analysis of the Terraform plan:

> Terraform apply stages for new planning.data.gov.uk subdomains
> 
>  * Delete existing Route 53 zone for development.digital-land.info
>  * Delete SSL cert for *.development.digital-land.info
>  * Create new Route 53 zone for development.planning.data.gov.uk
>  * Create new SSL cert validation records in new Route 53 zone
>  * Create new set of DNS records in new Route 53 zone
>  * Create SSL cert for *.development.planning.data.gov.uk
>  * Fixed response on load balancer listeners to be updated with new domain
>  * CDNs to be updated with new SSL cert and alias
>  * ECS tasks referencing domain in environments variables to be replaced - dependent services to be updated too
>  * MWAA to be updated with new SSL cert
>  * Airflow Load balancer to be updated with new SSL cert
>  * Update Cognito user pools with new domain pipelines.development.planning.data.gov.uk
>  * CloudWatch Dashboard (Overwatch) to be updated with new domain references
>  * Redirects to be re-created in new Route 53 zone  (SHOULD WE PRESERVE OLD ROUTE 53 ZONE FOR REDIRECTS?)
> 
>  Plan: 29 to add, 47 to change, 29 to destroy.

Would be good to stay after standup tomorrow morning and discuss.

**Comment by cpcundill** [comment]
Agreed with @Ben-Hodgkiss  and @eveleighoj that we want to avoid the complexity of preserving redirects from the current domain in dev and staging environments.

We'll need to agree a deployment window where some downtime should be expected - especially due to DNS replication delays.

@Ben-Hodgkiss will send message on Planning Data Team Slack channel to inform team and make sure there are no objections.

**Comment by cpcundill** [comment]
Migration of development environment is complete
